### PR TITLE
Re-theme the default layout and home page

### PR DESCRIPTION
**Pull request by amitaibu**
This is still WIP and PR Depends on #133, but I open it to see if there is enough interest, so I could invest more time.

The motivation for this PR is to improve the first impression devs get when scaffolding for the first time, and on the same time provide some of the typical functionality of a website (along with my non-hidden agenda is also to improve my Yesod skills by getting reviews 😉 )

Here's the [diff](https://github.com/amitaibu/yesod-scaffold/compare/auth-dummy...amitaibu:semantic-ui) until #133 is merged

joxi_screenshot_1473197789630

welcome_to_yesod_

### CODE REVIEW

File: Foundation.hs [hunk 1]
```diff
@@ -23,6 +25,16 @@ data App = App
     , appLogger      :: Logger
     }
 
+data MenuItem = MenuItem
```
Comment (amitaibu) [inline Foundation.hs]: what would be the best place to move those types?

**Comment by snoyberg** [comment]
Probably worth emailing the Yesod mailing list about it. From my experience, Bootstrap is still very popular, and likely what most people would want in the scaffold.

**Comment by amitaibu** [comment]
Sure, I can keep bootstrap, but still modernize the look + add the
functionality

**Comment by amitaibu** [comment]
I _tried_  to do it with Bootstrap. I mean, I've been using Bootstrap for years, but my problem with the different MIT licenses [themes](https://bootswatch.com/), that even-though I'm able to easily replace the css file and get a new look and feel, in my eyes it all kinda looks the same.

welcome_to_yesod_

What I like about [SUI](http://semantic-ui.com/) (semantic UI) is that:

1. It gives a fresh look (= admitedly by the fact it's not as popular as bootstrap), not the bootstrapy one we're so used to. This IMO is inline with showing yesod as a modern framework
1. It's a matter of taste of course, but I think that for prototyping SUI allows having nicer UIs

Anyway, I'll sleep on it. Lets see how I feel about it in the morning 😉

**Comment by amitaibu** [comment]
Ok, decided to stay with Bootstrap, so it would be easier to get it in 😄 

joxi_screenshot_1473444993635

File: templates/homepage.hamlet [hunk 2]
```diff
@@ -1,80 +1,141 @@
...
-    And last but not least, Testing. In <tt>test/Spec.hs</tt> you will find a #
-    test suite that performs tests on this page. #
-    You can run your tests by doing: <pre>stack test</pre>
+<div class="masthead">
+    <div class="container">
+        <div class="row">
+            <h1 class="header">
+                Yesod; A modern framework for blazing fast websites
+            <h2>
+                Fast, Stable & spiced with great community
```
Comment (MaxGabriel) [inline templates/homepage.hamlet]: It's weird how Fast/Stable are capitalized but "spiced with a great community" is not

File: templates/homepage.hamlet [hunk 3]
```diff
@@ -1,80 +1,141 @@
...
-
-  <p>
-    And last but not least, Testing. In <tt>test/Spec.hs</tt> you will find a #
-    test suite that performs tests on this page. #
-    You can run your tests by doing: <pre>stack test</pre>
+<div class="masthead">
+    <div class="container">
+        <div class="row">
+            <h1 class="header">
+                Yesod; A modern framework for blazing fast websites
```
Comment (MaxGabriel) [inline templates/homepage.hamlet]: Maybe just me, but "Yesod; A" feels a little weird to me. I guess because "Yesod" isn't a full sentence/capitalizing A after a semicolon isn't right

File: templates/homepage.hamlet (hunk 2 above)
Comment (amitaibu) [inline templates/homepage.hamlet]: Pushed a fix

**Comment by MaxGabriel** [comment]
This redesign looks really good, nice work @amitaibu.

File: templates/homepage.hamlet (hunk 3 above)
Comment (amitaibu) [inline templates/homepage.hamlet]: you mean you'd expect to have a lower case of the `a`?

Comment (MaxGabriel) [inline templates/homepage.hamlet]: I'm not really sure what I'm expecting. Maybe a colon instead of a semicolon?

**Comment by amitaibu** [comment]
Thanks @MaxGabriel  :)

There's one thing I know I wasn't consistent with - showing the classes in the divs - `<div .container>` vs `<div class="container">`. I actually prefer the latter, since it allows me to copy/ paste more easily.

We should probably be consistent, so I wonder if you folks agree with my comment above.

File: templates/homepage.hamlet (hunk 3 above)
Comment (amitaibu) [inline templates/homepage.hamlet]: The semicolon just looked kind of cool to me, but this can obviously change. I'm not a native English speaker, so I ben the English rules to my will 😜

File: templates/homepage.julius [hunk 4]
```diff
@@ -22,6 +22,7 @@ $(function() {
       success: function (data) {
         var newNode = $("<li></li>");
         newNode.text(data.message);
+        console.log(data);
         $("##{rawJS commentListId}").append(newNode);
```
Comment (psibi) [inline templates/homepage.julius]: @amitaibu You may want to remove the console part ?

**Comment by psibi** [comment]
Nice work @amitaibu . This looks very good. I do feel that the semicolon looks a bit odd as Maximilian told. Also, I think `<div .container>` is preferable as it demonstrates Shakespeare capabilities. IIRC, the old code followed that format, but I don't have strong opinions.

File: templates/homepage.julius (hunk 4 above)
Comment (amitaibu) [inline templates/homepage.julius]: this is actually a feature not a bug ;) I have added this [text](https://github.com/yesodweb/yesod-scaffold/pull/135/files#diff-a8900c26af8ddefc60773f874e4506c4R124), so devs can see the response object.

File: Foundation.hs (hunk 1 above)
Comment (amitaibu) [inline Foundation.hs]: Should this stay in `Foundation.hs` or is there a better place for it?

**Comment by amitaibu** [comment]
> Also, I think <div .container> is preferable as it demonstrates Shakespeare capabilities

It does, however IMO in terms of easily copy/ paste from other resources it's making it harder. But if that's what holds the PR I'll change it :)

**Comment by chreekat** [comment]
On Mon, Sep 12, 2016 at 12:03:50PM -0700, Amitai Burstein wrote:
> > Also, I think <div .container> is preferable as it demonstrates
> > Shakespeare capabilities
> 
> It does, however IMO in terms of easily copy/ paste from other
> resources it's making it harder. But if that's what holds the PR
> I'll change it :)

I've come to think that Hamlet should not be considered as something
that is copy/pastable from other resources. :) Not even <pre> blocks
can be copy/pasted.

So why not

    <.container>

which also expands to <div class="container">.

**Comment by amitaibu** [comment]
* I have replaced the `;` with an em-dash
* Converted HTML to use Shakespeare

joxi_screenshot_1473876778265

Not sure why tests are failing, not sure it's related to this PR

File: test/Handler/HomeSpec.hs [hunk 5]
```diff
@@ -9,7 +9,7 @@ spec = withApp $ do
         it "loads the index and checks it looks right" $ do
           get HomeR
           statusIs 200
-          htmlAllContain "h1" "Welcome to Yesod"
+          htmlAnyContain "h1" "Yesod; A modern framework for blazing fast websites"
```
Comment (psibi) [inline test/Handler/HomeSpec.hs]: I think the test will fail here because it still contains `;` but you have changed the content.

Comment (amitaibu) [inline test/Handler/HomeSpec.hs]: Correct, I have fixed it. The tests still fail, but on what seems to look like an unrelated issue.

**Comment by amitaibu** [comment]
Update: I see there's indeed one test that fails that I'll fix, but in [one environment](https://travis-ci.org/yesodweb/yesod-scaffold/jobs/160123518) the test even doesn't start

**Comment by amitaibu** [comment]
`stack test` is passing
build__260_-_yesodweb_yesod-scaffold_-_travis_ci

 but the other one fails due to wrong dependency

job__260_1_-_yesodweb_yesod-scaffold_-_travis_ci

**Comment by amitaibu** [comment]
@psibi do you happen to know how I can solve this? ^^

**Comment by psibi** [comment]
The error seems to be happening because the latest lts uses
ghc-8.0.1. And we have restricted the use of base-4.9.0.0 in our cabal
file because of this bug: https://ghc.haskell.org/trac/ghc/ticket/12130

So, we have to wait till ghc 8.0.2 is released. Probably, till that we
can include that build in the allow_failures section with a proper
comment ?

**Comment by snoyberg** [comment]
This is correct. I can update the branches now, and then merging in the changes should unblock you @amitaibu.

**Comment by snoyberg** [comment]
OK, changes pushed, Travis will hopefully accept them.

**Comment by amitaibu** [comment]
Thanks @snoyberg , re-rolled and tests have passed. So I guess it's ready for review 😄

**Comment by psibi** [comment]
LGTM. CC: @MaxGabriel and others

**Comment by psibi** [comment]
Thanks!

**Comment by amitaibu** [comment]
Awesome!
